pkg/gather/common/common.go

```diff
@@ -0,0 +1,35 @@
+package common
+
+import (
+	"context"
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// GatheringFunction describes a gathering function
+type GatheringFunction struct {
+	CanFail  bool
+	Function interface{}
+}
+
+// FailableFunc creates an object describing a gathering function that canNOT fail
+func ImportantFunc(function interface{}) GatheringFunction {
+	return GatheringFunction{
+		CanFail:  false,
+		Function: function,
+	}
+}
+
+// FailableFunc creates an object describing a gathering function that can fail
+func FailableFunc(function interface{}) GatheringFunction {
+	return GatheringFunction{
+		CanFail:  true,
+		Function: function,
+	}
+}
```

Comment:
These three (`GatheringFunction`, `ImportantFunc`, `FailableFunc`) seem to be used in the clusterconfig gatherer. Maybe it would make sense to move them there and change the `GatheringFunction` to:
```
type GatheringFunction struct {
	CanFail  bool
	Function GathererFuncPtr
}
```